pkg/vm/engine/tae/iface/data/block.go

```diff
@@ -48,6 +48,8 @@ type BlockAppender interface {
 	) (int, error)
 	IsAppendable() bool
 	ReplayAppend(bat *containers.Batch) error
+	Ref()
```

Comment:
Don't use Ref and Unref.
Add a Close() for appended and the user program need to `Close` the appended explicitly.